test/utils.test.js

```diff
@@ -98,6 +98,38 @@ describe('utils', () => {
     expect(utils.isPluginOfPackage(base, other)).to.be.false;
   });
 
+  it('should check whether or not its lib is a plugin of another package', () => {
+    var base = {
+      name: 'parent',
+      build: {
+        type: 'app'
+      }
+    };
+
+    var lib = {
+      name: 'chai',
+      build: {
+        type: 'lib'
+      }
+    };
+
+    var libPlugin = {
+      name: 'chai-plugin',
+      build: {
+        type: 'plugin'
+      }
+    };
+
+    var depStack = ['parent', 'chai', 'chai-plugin'];
+    // Seems like the only place we actually get the package is with chai.
+    // Let the actual code work, rather than changing it for the test to work
+    utils.getPackage('chai')['build'] = {
+      'type': 'lib'
+    };
```

Comment:
`utils.getPackage` wraps the node.js `require()`, which caches things it loads, meaning that anything else which subsequently runs `require('chai')` will pick this up. While it works, I think we should use a better example for future tests. Let's consider adding the [`chai-spies` plugin](https://www.chaijs.com/plugins/chai-spies/) and use a spy instead.

Reply:
yah i forgot i did that, really the issue was it is requesting an actual package. Not sure how chai-spies would make a difference for this case? or you saying refactor the whole test suite to use chai-spies instead of chai? This test really just needed to be able to request a real package that has the json "build":{type:"lib"}

Reply:
> really the issue was it is requesting an actual package

Correct. That's why I'm suggesting that it use a spy. That way it can return the mock package declared above, and it will do that only in the test that is mocking it. If you would prefer using real files instead, then perhaps a mock workspace/project setup should be used like we do for many of the other tests.